frontend/src/utilities/__tests__/useDraggableTable.spec.ts

```diff
@@ -0,0 +1,123 @@
+        currentTarget: {
+          id: 'item3',
+          classList: { add: jest.fn() },
+          setAttribute: jest.fn(),
+        },
+      } as unknown as React.DragEvent<HTMLTableRowElement>);
+    });
+    expect(renderResult).hookToHaveUpdateCount(2);
+    expect(renderResult.result.current.tableProps.className).toBe('pf-m-drag-over');
+
+    // Simulate drag over
+    const { onDragOver } = renderResult.result.current.tableProps.tbodyProps;
+    act(() => {
+      onDragOver({
+        preventDefault: jest.fn(),
+        target: bodyRef.children[0],
+      } as unknown as React.DragEvent<HTMLTableSectionElement>);
+    });
+
+    expect(renderResult).hookToHaveUpdateCount(3);
+    expect(renderResult.result.current.tableProps.className).toBe('pf-m-drag-over');
+
+    // Simulate drag end
+    const { onDragEnd } = renderResult.result.current.rowProps;
+    act(() => {
+      onDragEnd({
+        target: {
+          id: 'item4',
+          classList: { remove: jest.fn() },
+          setAttribute: jest.fn(),
+        },
+      } as unknown as React.DragEvent<HTMLTableRowElement>);
+    });
+
+    expect(renderResult).hookToHaveUpdateCount(4);
+    expect(renderResult.result.current.tableProps.className).toBe(undefined);
+
+    // Simulate drop event
+    const { onDrop } = renderResult.result.current.rowProps;
+    renderResult.result.current.tableProps.tbodyProps.ref.current = bodyRef;
+    renderResult.result.current.tableProps.tbodyProps.ref.current.getBoundingClientRect = jest
+      .fn()
+      .mockReturnValue({
+        x: 40,
+        y: 40,
+        width: 20,
+        height: 20,
+      } as DOMRect);
+
+    act(() => {
+      onDrop({
+        clientX: 45,
+        clientY: 45,
+      } as unknown as React.DragEvent<HTMLTableRowElement>);
+    });
+    expect(setItemOrder).toHaveBeenCalledWith(['item3', 'item1', 'item2']);
+
+    // Test isValidDrop when drop is not valid
+    renderResult.result.current.tableProps.tbodyProps.ref.current.getBoundingClientRect = jest
+      .fn()
+      .mockReturnValue({} as DOMRect);
+    act(() => {
+      onDrop({
+        clientX: 45,
+        clientY: 45,
+      } as unknown as React.DragEvent<HTMLTableRowElement>);
+    });
+    expect(setItemOrder).toHaveBeenCalledWith(['item3', 'item1', 'item2']);
+    expect(renderResult.result.current.tableProps.className).toBe(undefined);
+  });
+
+  it('should revert item order if drag leaves the drop zone', () => {
+    const itemOrder = ['item1', 'item2', 'item3'];
+    const bodyRef = generateRowsWithItems(itemOrder);
+    const renderResult = testHook(useDraggableTable)(itemOrder, setItemOrder);
+    renderResult.result.current.tableProps.tbodyProps.ref.current = bodyRef;
+
+    const { onDragLeave } = renderResult.result.current.tableProps.tbodyProps;
+
+    // Simulate drag leaving drop zone
+    act(() => {
+      renderResult.result.current.tableProps.tbodyProps.ref.current = null;
```

Comment:
You just set this value in the line before and not you are resetting it. Don't set the value in the first place if you don't need it. Although I would expect the ref is required for drag events to be processed properly.